weldx/core.py

```diff
@@ -289,8 +289,8 @@ def __init__(
             'step', 'linear'.
 
         """
-        self._data = None
-        self._time_var_name = None
+        self._data = None  # type: Union[MathematicalExpression, xr.DataArray]
```

Comment:
why do you choose such a format here? You could directly attach the type annotation to the variable, right?

Reply:
```suggestion
        self._data: Union[MathematicalExpression, xr.DataArray] = None
```

Reply:
Maybe. Works for dataclasses. Not sure if it works for local vars too